docs/source/reproducible.rst

```diff
@@ -104,3 +104,47 @@ configuration file and use it the script like so:
         # Set the seed for numpy
         np.random.seed(config.seed)
         # application-specific seed setting
+
+Instantiating objects
+---------------------
+
+Another cool feature of Hydra is `object instantiating <https://hydra.cc/docs/advanced/instantiate_objects/overview/>`_.
+Imagine you want to try different optimizers for your Deep Neural Network (DNN) or you want to try different DNNs in the same pipeline.
+Instead of doing ``if-else`` statements, you write one line of code and let Hydra choose the
+appropriate object class based on your configuration. See the script
+`examples/real_convolve.py <https://github.com/ebezzam/python-dev-tips/blob/main/examples/real_convolve.py>`_
+for the example.
+
+.. code-block:: python
+
+    @hydra.main(version_base=None, config_path="configs", config_name="defaults")
+    def main(config):
+        # instantiate object from config
+        signal = instantiate(config.signal)
+        # application specific choice of object class
+
+``instantiate`` function from ``hydra.utils`` allows you to define an object in a YAML file 
+without being tied to a particular class. To do this, you need to define ``_target_`` in 
+your config (see configs in ``configs/signal``) and object initialization arguments. Object class 
+can be either defined in your project (``configs/signal/ExampleZeros``, ``configs/signal/ExampleCustom``)
+or taken from a package (``configs/signal/ExampleNumpy``).
+
+Note that here we use another Hydra feature: config grouping and splitting. Instead of writing 
+configurations for all objects in the main config and copying configuration files, we create a sub-directory ``signal``,
+where all ``signal`` configs are defined. Now we can run the main config with the ``signal`` of
+our choice simply by specifying it in the command line. For example, ``python real_convolve.py signal=ExampleNumpy``
+or ``python real_convolve.py signal=ExampleZeros``.
```

Comment:
can you make file path "scripts/real_convolve.py" (to assume people run from repo root)